Create IdV service

ertzeid opened the pull request:
Create a mock ID Verification service with tests.

cc @jgrevich @sashax

sashax commented:
Hey, Amanda, this looks great. Can you add instructions to run the app to the readme, too?

And then if you squash the commits, I'll be happy to merge it!

jgrevich commented on app/app.rb:
```diff
@@ -0,0 +1,86 @@
+    'Hello, USA!'
+  end
+
+  get '/idvaas/user/:user_uuid/question/?' do
```
I don't believe we need the trailing `?`. For this first iteration it should be sufficient to return a single question that makes the pass/fail route obvious. It also needs to match the routes used by the IdP:

```
  def question
    response = IdpRequest.new("/user/#{current_user.uuid}/question/").response
    ...
  end

  def answer
    response = IdpPostRequest.new("/user/#{current_user.uuid}/question/", answer_data).response
    ...
  end
```

jgrevich replied:
Ahh, I see what you were doing (http://www.sinatrarb.com/faq.html#slash). Good call!

jgrevich commented on app/app.rb:
```diff
@@ -0,0 +1,86 @@
+    }.to_json
+  end
+
+  get '/idvaas/user/:user_uuid/?' do
```
we can also remove the `/idvaas/` from all of these since it will be hosted on its own domain: `projectName-idv.tld`

jgrevich commented on test/app_test.rb:
```diff
@@ -0,0 +1,68 @@
+         '{"questionId":"eaef8225-0eed-4cde-817b-f1850302b7bc","key":5}'
+    assert last_response.ok?
+    assert_equal '{"quizStatus":"PASSED"}', last_response.body
+  end
```
Looks good other than what's mentioned in the comments above.

We may want to look into https://github.com/brynary/rack-test down the road.

jgrevich replied:
Doh, I missed this: https://github.com/18F/identity-idv/pull/1/files#diff-8b7db4d5cc4b8f6dc8feb7030baa2478R3

ertzeid committed:
Create mock ID Verification service

ertzeid commented:
Fixed based on all comments; squashed; force-pushed.

Have you ever tried `git merge --squash`? I've never tried, but heard it can be useful (though it might not work from the GitHub UI directly)

jgrevich commented:
awesome possum! :shipit:
